### PR TITLE
Add the ability to search for organizations by their attributes

### LINKED ISSUE #88: Adding keycloaks current filter options to api-endpoints
Hey @xgp,
would it be possible to add filter options like "q" f.ex. for the attribute based filtering to the existing api-endpoints?
- q :: A query to search for custom attributes, in the format 'key1:value2 key2:value2'
- <param-name> :: for queriing fields of the organization

It would be realy nice to have the same syntax of queriing data from your extensions, that from keycloak itself.

tank you very much.

### PR DESCRIPTION
**Pull request by phamann**
### TL;DR
Following on from discussion in #88, this adds the ability to search for organizations based on their key=value attribute pairs, using a new `q` query paramter on the search resource, i.e `GET /:realm/orgs?q=foo:bar`.

### Notes:
- The `q` and `key:value` search param follows convention in Keycloak core, such as the [User search API](https://github.com/keycloak/keycloak/blob/main/services/src/main/java/org/keycloak/services/resources/admin/UsersResource.java#L231-L338).
- ~~The API will now return a `400 Bad request` if both a `search` and `q` parameter are provided.~~ 
- I haven't included the ability to query other organization fields other than attirbutes (as there aren't many and my personal need was for attributes). However, it should be an easy refactor in the future.
- As this addition isn't yet avalable via the phase-two java client, I had to add a raw HTTP test, I hope this is ok?
- As this is a new optional query param, I consider this a non-breaking change to the API, but let me know if you disagree. 

---

Closes #88

### CODE REVIEW

**Review by xgp — APPROVED** [review]
This looks great. We can refactor the tests later, as there is always the problem of not having the java client ready for the tests in the rev we make the api changes.

**Comment by xgp** [comment]
@phamann Thanks so much for this addition.

Regarding your tests question, this isn't a problem, as the phasetwo-java lib isn't updated until after we make API changes. However, it (and the other API client libs) gets generated based on a spec file here: https://github.com/p2-inc/phasetwo-docs/blob/master/openapi.yaml . Would you mind doing a PR for that spec file for the change you made?

**Comment by phamann** [comment]
Hey @xgp, I've now submitted a PR to also update the OpenAPI schema as requested. https://github.com/p2-inc/phasetwo-docs/pull/94

File: src/main/java/io/phasetwo/service/resource/OrganizationsResource.java [hunk 1]
```diff
@@ -68,13 +70,31 @@ public Response me() {
...
       @QueryParam("first") Integer firstResult,
-      @QueryParam("max") Integer maxResults) {
-    log.debugf("listOrgs %s %s %d %d", realm.getName(), searchQuery, firstResult, maxResults);
+      @QueryParam("max") Integer maxResults,
+      @QueryParam("q") String searchQuery) {
     firstResult = firstResult != null ? firstResult : 0;
     maxResults = maxResults != null ? maxResults : Constants.DEFAULT_MAX_RESULTS;
-    return orgs.searchForOrganizationByNameStream(realm, searchQuery, firstResult, maxResults)
+
+    if (search != null && searchQuery != null) {
```
Comment (xgp) [inline src/main/java/io/phasetwo/service/resource/OrganizationsResource.java:80]: @phamann Thinking more about this as I update the java client. Would there be a use case where the caller would expect if both were present that it should search on both an attribute **and** a name?

Comment (xgp) [reply src/main/java/io/phasetwo/service/resource/OrganizationsResource.java]: Thanks for doing that @phamann . We are aware of some customer use cases where they would probably use both.

**Comment by paulwer** [comment]
@xgp when did you plan to release this in the docker oss environment / the client sdk?

**Comment by xgp** [comment]
- client went out a few days ago https://repo1.maven.org/maven2/io/phasetwo/phasetwo-admin-client/0.1.4/
- docker image will be updated before eow

**Comment by paulwer** [comment]
@xgp i didnt see any updates at keycloak-js did I miss something? or is this still pending?

**Comment by xgp** [comment]
Sorry, the java client went out. js client still pending.

**Comment by xgp** [comment]
@paulwer these are released.
